cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -0,0 +1,187 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/systemd"
+)
+
+var (
+	timeNow = time.Now
+
+	// default 1:30, as that is how long systemd will wait for services by
+	// default so seems a sensible default
+	defaultMountUnitWaitTimeout = time.Minute + 30*time.Second
+
+	unitFileDependOverride = `[Unit]
+Requires=%[1]s
+After=%[1]s
+`
+
+	doSystemdMount = doSystemdMountImpl
+)
+
+// SystemdMountOptions reflects the set of options for mounting something using
+// systemd-mount(1)
+type SystemdMountOptions struct {
+	// IsTmpfs indicates that "what" should be ignored and a new tmpfs should
+	// be mounted at the location.
+	IsTmpfs bool
+	// EphemeralInitramfsMount indicates that the mount should not persist from
+	// the initramfs to after the pivot_root to normal userspace. The default
+	// value, false, means that the mount will persist across the transition,
+	// this is done by creating systemd unit overrides for various initrd
+	// targets in /run that systemd understands when it isolates to the
+	// initrd-cleanup.target when the pivot_root is performed.
+	EphemeralInitramfsMount bool
```

Comment:
given the context this is in, this can maybe be slightly shorter:

* Ephemeral
* NotAfterInitramfs
* EphemeralForInitramfs

Reply:
since this is currently only used for the initramfs, probably it's fine to just use `Ephemeral`, if we ever end up using this function outside of the Initramfs we would need to rename, but then the whole implementation of the function will need to change too so that's probably not an issue